src/features/Router/Router.tsx

```diff
@@ -0,0 +1,57 @@
+import { createBrowserRouter, Params, RouterProvider } from 'react-router-dom';
+import { Main } from '../../pages/Main/Main';
+import { ErrorPage } from '../../pages/ErrorPage/ErrorPage';
+import { DetailedCard } from '../DetailedCard/DetailedCard';
+import { mainLoader } from './Loaders/mainLoader';
+import { detailedLoader } from './Loaders/detailedLoader';
+import { useLocalStorage } from '../../hooks/useLocalStorage';
+import { search } from './Actions/search';
+import { RouterParams, RouterPath } from './Router.enum';
+import { useCallback } from 'react';
+
+export function Router() {
+  const { searchValue, setSearchValue } = useLocalStorage();
+
+  const rootLoader = useCallback(async () => mainLoader(searchValue, 0), [searchValue]);
+  const pageLoader = useCallback(
+    async ({ params }: { params: Params<RouterParams.PAGE> }) =>
+      mainLoader(searchValue, Number.parseInt(params.page ?? '0')),
+    [searchValue]
+  );
+  const searchAction = useCallback(
+    async ({ request }: { request: Request }) => search({ request }, setSearchValue),
+    [setSearchValue]
+  );
+
+  const router = createBrowserRouter([
+    {
+      path: '/',
+      element: <Main searchValue={searchValue} />,
+      errorElement: <ErrorPage />,
+      loader: rootLoader,
```

Comment:
I dont like this `useLoaderData` at all, to be honest. it's not safe at all, you use type assertion all the time. if you change the names of properties in your loader, your application will crash, but not the types in the components, but the whole point of the typescript is exactly the opposite. I never used this, but I guess what can be done here at least is to create a separate wrapper hook, which will at least regulate types by loader
```typescript
export const useMainLoaderData = () => {
  return useLoaderData() as Awaited<ReturnType<typeof mainLoader>>;
};
```
and create a new for for each new loader. Or you can play with generic types to make it universal